packages/oidc/src/dpop/tokenExchange.ts

```diff
@@ -88,17 +89,11 @@ function hasExpiresIn(
   return value.expires_in === undefined || typeof value.expires_in === "number";
 }
 
-export type TokenEndpointResponse = {
-  accessToken: string;
+export type CodeExchangeResult = TokenEndpointResponse & {
+  // The idToken must not be undefined after the auth code exchange
   idToken: string;
   webId: string;
-  refreshToken?: string;
   dpopKey?: KeyPair;
-  expiresIn?: number;
-};
-
-export type TokenEndpointDpopResponse = TokenEndpointResponse & {
-  dpopKey: KeyPair;
 };
```

Comment:
Is this no longer useful?

Reply:
Not really: I extended the `fetchFactory` so that it exports a single function (`buildAuthenticatedFetch`) which accepts a `KeyPair | undefined` rather than the factory exporting two functions, `buildBearerFetch` and `buildDpopFetch`. This way, the type checking is done in the core module only, rather than in each environment-specific module.

Reply:
Gotcha, makes sense :+1: